moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1636,33 +1634,36 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::Attache
       if (object.object.operation == moveit_msgs::CollisionObject::ADD ||
           !robot_state_->hasAttachedBody(object.object.id))
       {
-        if (robot_state_->clearAttachedBody(object.object.id))
+        if (robot_state_->clearAttachedBody(object_to_attach.id_))
           ROS_DEBUG_NAMED(LOGNAME, "The robot state already had an object named '%s' attached to link '%s'. "
                                    "The object was replaced.",
-                          object.object.id.c_str(), object.link_name.c_str());
+                          object_to_attach.id_.c_str(), object.link_name.c_str());
 
-        robot_state_->attachBody(object.object.id, shapes, poses, object.touch_links, object.link_name,
-                                 object.detach_posture, subframe_poses);
-        ROS_DEBUG_NAMED(LOGNAME, "Attached object '%s' to link '%s'", object.object.id.c_str(),
+        robot_state_->attachBody(object_to_attach.id_, object_to_attach.shapes_, object_to_attach.shape_poses_,
+                                 object.touch_links, object.link_name, object.detach_posture,
+                                 object_to_attach.subframe_poses_);
+        ROS_DEBUG_NAMED(LOGNAME, "Attached object '%s' to link '%s'", object_to_attach.id_.c_str(),
                         object.link_name.c_str());
       }
+      // STEP 3.5: For append operations, only update the attached object without looking for it in the world
```

Comment:
This comment should actually have stayed in the other PR. Got to remember cherrypicking them.